rust/operator-binary/src/util.rs

```diff
@@ -0,0 +1,59 @@
+use snafu::{OptionExt, Snafu};
+use stackable_airflow_crd::AirflowCluster;
+use stackable_operator::k8s_openapi::api::batch::v1::Job;
+use stackable_operator::k8s_openapi::api::core::v1::{EnvVar, EnvVarSource, SecretKeySelector};
+
+#[derive(Snafu, Debug)]
+#[allow(clippy::enum_variant_names)]
+#[snafu(context(suffix(false)))]
+pub enum Error {
+    #[snafu(display("object defines no version"))]
+    ObjectHasNoVersion,
+}
+
+pub enum JobState {
+    InProgress,
+    Complete,
+    Failed,
+}
+
+pub fn get_job_state(job: &Job) -> JobState {
+    let conditions = job
+        .status
+        .as_ref()
+        .and_then(|status| status.conditions.clone())
+        .unwrap_or_default();
+
+    if conditions
+        .iter()
+        .any(|condition| condition.type_ == "Failed" && condition.status == "True")
+    {
+        JobState::Failed
+    } else if conditions
+        .iter()
+        .any(|condition| condition.type_ == "Complete" && condition.status == "True")
+    {
+        JobState::Complete
+    } else {
+        JobState::InProgress
+    }
+}
+
+pub fn airflow_version(airflow: &AirflowCluster) -> Result<&str, Error> {
+    airflow.spec.version.as_deref().context(ObjectHasNoVersion)
+}
+
+pub fn env_var_from_secret(var_name: &str, secret: &str, secret_key: &str) -> EnvVar {
+    EnvVar {
+        name: String::from(var_name),
+        value_from: Some(EnvVarSource {
+            secret_key_ref: Some(SecretKeySelector {
+                name: Some(String::from(secret)),
+                key: String::from(secret_key),
+                ..Default::default()
+            }),
+            ..Default::default()
+        }),
+        ..Default::default()
+    }
+}
```

Comment:
This is included in the `ContainerBuilder` via `add_env_var_from_secret` as i learned recently!

Reply:
Ah, good point! Though I've looked at the code again and I think it may be less readable if I use `add_env_var_from_secret` calls inline with a container, instead of defining them in a separately. Is it ok with you if I leave it like it is?

Reply:
This function is duplicated in other operators too because env vars get constructed from many sources and there is not always a `ContainerBuilder` around at that point. So I suggest we add this to `operator-rs` regardless of the `ContainerBuilder::add_env_var_from_secret()` in the future.